### ISSUE TITLE
Modal response for actions

### ISSUE THREAD

**Issue by samdevbr** [laravel/nova-issues#1504]
Sometimes i've to show some info after running some action, for example: generated username, oauth client id, and etc.

Would be great if after running an action, there was an modal response for actions, instead of simple toast messages.

**Comment by rcschettino** [comment]
+1

**Comment by davidhemphill** [comment]
Would be a great PR if you're up for it. Sounds handy. 🤙 

Hey there! In order to keep this repository focused on bug reports, we auto-close feature requests and requests for help. Feel free to post your feature requests so others can discuss and add reactions. We'll keep an eye on them for later planning.

**Comment by Grayda** [comment]
Has any progress been made on this? I'm trying to generate API keys using Sanctum and because I can't retrieve a plaintext key after it's been generated, I need to present it to the user.

**Comment by Conaire** [comment]
Has there been any updates on this?

**Comment by goldmerc** [comment]
+ 1 this would be a great feature

**Comment by f-liva** [comment]
+100

**Comment by Dontorpedo** [comment]
+1

**Comment by trippo** [comment]
+1

**Comment by crynobone** [comment]
My concern regarding this is that modal is just one type of response, and what if;

* we need to customize it even further?
* or, modal for copying response from action or message is not the only possible output?

I was wondering whether we can have a generic `emit` event fired directly from Action and JavaScript handles the rest. E.g:

```php
Action::emit('show-copy-modal', [
     'text' => 'Your new access token',
]); 
```

What's I not sure right now is whether the response modal should be part of Laravel Nova or can be offloaded to external packages. And if the later would it be enough to manage the requirement.

**Comment by crynobone** [comment]
This has been implemented in v3.21.0

See https://nova.laravel.com/docs/3.0/actions/defining-actions.html#action-response-modals for details.

**Comment by Conaire** [comment]
yay 🙏

________________________________

**Comment by schrapel** [comment]
> This has been implemented in v3.21.0
> 
> See https://nova.laravel.com/docs/3.0/actions/defining-actions.html#action-response-modals for details.

Thank you. What do you think about adding the possibility to open another action as well as an asset? For example we have an action to disable a user. Sometimes people want to send a user an email when performing this action, but we do not want to clutter the action with additional fields for sending the email. So we would like to put a boolean field that will choose whether to load a another action after the 1st was completed successfully. 

Right now we can achieve it by either adding these fields to dozens of actions that may require a email or redirect them to a new page where they can send the email

**Comment by rikiless** [comment]
Where to place following?

```js
import ApiTokenCopier from './components/ApiTokenCopier'

Nova.booting(Vue => {
  Vue.component('api-token-copier', ApiTokenCopier)
})
```
